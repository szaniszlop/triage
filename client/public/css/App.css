.App {
  text-align: left;
  display : flex;
  flex-direction: column;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  min-height: 8vh;
  max-height: 15vh;
  align-items: center;
  justify-content: center; 
  flex-wrap: nowrap; 
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background: white;
  padding: 5px;  
}

.App-header-logo {
  justify-content: left;
}

.App-header-title {
  justify-content: center;
  align-items: center;
  text-align: center;
}

.App-header-user-info {
  justify-content: right;
  padding: 5px;
  margin-right: 5px;
}

.App-header-logo img {
  height: 30px;
  padding: inherit;
}

.App-header-user-info img {
  height: 30px;
  border-radius: 50%;
  padding: inherit;
}

.App-header-user-info div {
  padding: inherit;
}

.App-content{
  min-height: 80vh;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: left;
  float: left;
  margin-top: 15vh;
  margin-left: 10px;
  margin-right: 5px;
}

.App-item-details-box {
  margin-left: 5px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.App-nav{
  min-height: 2vh;
  max-height: 5vh;
  align-items: center;
  justify-content: center;
  background: white;
  padding: 5px;    
}

.App-tabstop{
  display: inline-block;
  margin: 2px 2px 0 0;
}

.App-error{
  color: crimson;
  background-color: lightyellow;
  align-items: center;
  justify-content: center;  
  text-align: center;
}

.App-info{
  color: rgb(50, 20, 220);
  background-color: lightyellow;
  align-items: center;
  justify-content: center;  
  text-align: center;
}

.App-codebox {
  border-width:2px;
  border-style:solid;
  border-color:black;
}

button, [type=link]{
  background: transparent;
  padding: 5px;
  border: none;
  display: flex;
  margin: 5px;
}

.button:hover, [type=link]:hover {
  color: #0250bc;
  background: transparent;
  border: none;
}

