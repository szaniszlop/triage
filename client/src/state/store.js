import { configureStore } from '@reduxjs/toolkit'

import userReducer from './userSlice'
import invoicesReducer from './invoiceSlice'
import navigationReducer from './navigationSlice'
import tenantReducer from './tenantSlice'

export default configureStore({
  reducer: {
    user: userReducer,
    invoices: invoicesReducer,
    navigation: navigationReducer,
    tenants: tenantReducer
  }
})