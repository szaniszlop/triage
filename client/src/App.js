import React, {useEffect} from 'react'
import {Outlet} from 'react-router-dom'
import Header from './components/application/headerComponent.js'
import { useDispatch, useSelector } from 'react-redux'
import {useAuth0} from '@auth0/auth0-react'
import {userLoggedIn} from './state/userSlice';
import { selectLocation } from './state/navigationSlice';
import { useNavigate } from 'react-router-dom';
import { setNavigation } from './state/navigationSlice.js'

function App() {
  console.log("inside App");
  const { isAuthenticated, user } = useAuth0();    
  const dispatch = useDispatch();
  const navigation = useNavigate();
  const location = useSelector(selectLocation);

  useEffect(() =>{ 
    console.log("App effect", isAuthenticated, user);
    if(isAuthenticated){
      dispatch(userLoggedIn(user));
      if(location && location.location && location.location.pathname){
        console.log("App effect dispatch to location", location);
        navigation(location);
        dispatch( setNavigation({}));
      }
    }
  }, [isAuthenticated, dispatch, user, navigation, location])

  return (
    <div className="App">
      <Header/>
      <div className="App-content">
        <Outlet />
      </div>
    </div>
  );
}

export default App;
