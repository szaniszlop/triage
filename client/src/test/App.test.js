import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import store from '../state/store';

test('renders application header', () => {
  const component = renderer.create(
  <MemoryRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </MemoryRouter>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
