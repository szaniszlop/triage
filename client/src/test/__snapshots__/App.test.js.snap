// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders application header 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="flex-row"
    >
      <div
        className="flex-large one-third App-header-logo"
      >
        <div>
          <img
            alt="logo"
            className=".App-logo"
            src="/images/logo.svg"
          />
        </div>
      </div>
      <div
        className="flex-large one-third App-header-title"
      >
        <h1>
          Test app
        </h1>
      </div>
      <div
        className=" flex row App-header-user-info"
      >
        <button
          onClick={[Function]}
          type="link"
        >
          Login
        </button>
         
          
      </div>
    </div>
    <nav
      className="flex-row App-nav"
    >
      <a
        className="App-tabstop round-button button"
        href="/about"
        onClick={[Function]}
      >
         
        About
         
      </a>
    </nav>
  </header>
  <div
    className="App-content"
  />
</div>
`;
